languagetool-language-modules/es/src/main/java/org/languagetool/language/Spanish.java

```diff
@@ -196,85 +196,109 @@ public LanguageMaintainedState getMaintainedState() {
+    id2prio.put("POR_CIERTO", 30);
+    id2prio.put("DEGREE_CHAR", 30); // greater than SPACE_UNITIES
+    id2prio.put("LO_LOS", 30);
+    id2prio.put("ETCETERA", 30); // greater than other typography rules
+    id2prio.put("P_EJ", 30); // greater than other typography rules
+    id2prio.put("SE_CREO2", 25); 
+    //id2prio.put("ESPACIO_DESPUES_DE_PUNTO", 25); // greater than other typography rules
+    id2prio.put("AGREEMENT_ADJ_NOUN_AREA", 30); // greater than AGREEMENT_DET_NOUN
+    id2prio.put("PRONOMBRE_SIN_VERBO", 25); // inside CONFUSIONS, but less than other rules ?
+    id2prio.put("AGREEMENT_DET_ABREV", 25); // greater than AGREEMENT_DET_NOUN
+    id2prio.put("MUCHO_NF", 25); // greater than AGREEMENT_DET_NOUN
+    id2prio.put("AGREEMENT_DET_NOUN_EXCEPTIONS", 25); // greater than AGREEMENT_DET_NOUN
+    id2prio.put("TYPOGRAPHY", 20); // greater than AGREEMENT_DET_NOUN
+    id2prio.put("AGREEMENT_DET_NOUN", 15);
+    //id2prio.put("PRONOMBRE_SIN_VERBO", 20);
+    id2prio.put("AGREEMENT_DET_ADJ", 10);
+    id2prio.put("CONFUSION_ES_SE", 20); //lower than diacrtics rules
+    id2prio.put("HALLA_HAYA", 10);
+    id2prio.put("VALLA_VAYA", 10);
+    id2prio.put("SI_AFIRMACION", 10); // less than DIACRITICS
+    id2prio.put("TE_TILDE2", 10); // less than PRONOMBRE_SIN_VERBO
+    id2prio.put("SEPARADO", 1);
+    id2prio.put("ES_SPLIT_WORDS", -10);
+    id2prio.put("U_NO", -10);
+    id2prio.put("EL_TILDE", -10);
+    id2prio.put("SINGLE_CHARACTER", -15); // less than ES_SPLIT_WORDS
+    id2prio.put("TOO_LONG_PARAGRAPH", -15);
+    id2prio.put("PREP_VERB", -20);
+    id2prio.put("SUBJUNTIVO_FUTURO", -30);
+    id2prio.put("SUBJUNTIVO_PASADO", -30);
+    id2prio.put("SUBJUNTIVO_PASADO2", -30);
+    id2prio.put("AGREEMENT_ADJ_NOUN", -30);
+    id2prio.put("AGREEMENT_PARTICIPLE_NOUN", -30);
+    id2prio.put("AGREEMENT_POSTPONED_ADJ", -30);
+    id2prio.put("MULTI_ADJ", -30);
+    id2prio.put("SUBJUNTIVO_INCORRECTO", -40);
+    id2prio.put("COMMA_SINO", -40);
+    id2prio.put("COMMA_SINO2", -40);
+    id2prio.put("VOSEO", -40);
+    id2prio.put("REPETITIONS_STYLE", -50);
+    id2prio.put("MORFOLOGIK_RULE_ES", -100);
+    id2prio.put("PHRASE_REPETITION", -150);
+    id2prio.put("SPANISH_WORD_REPEAT_RULE", -150);
+    id2prio.put("UPPERCASE_SENTENCE_START", -200);
+    id2prio.put("ES_QUESTION_MARK", -250);	  
+  }
+
+  @Override
+  public Map<String, Integer> getPriorityMap() {
+    return id2prio;
+  }
+  
   @Override
   protected int getPriorityForId(String id) {
+    if (id.equals("CONFUSIONS2")) {
+      return 50; // greater than CONFUSIONS
+    }
+    if (id.equals("RARE_WORDS")) {
+      return 50;
+    }
+    if (id.equals("MISSPELLING")) {
+      return 40;
+    }
+    if (id.equals("CONFUSIONS")) {
+      return 40;
+    }
+    if (id.equals("INCORRECT_EXPRESSIONS")) {
+      return 40;
+    }
+    if (id.equals("DIACRITICS")) {
+      return 30;
+    } 
```

Comment:
Not sure about these - shouldn't all IDs be in the map, except for those with a `startsWith`?

Reply:
Those are category IDs. Moving them to the map leads to ID-not-found-error